### ISSUE TITLE
HomeConnect API unexpected values (TQ907DF5/01)

### ISSUE THREAD

**Issue by Mibra-de** [thoukydides/homebridge-homeconnect#209]
### Plugin Version

0.37.6

### Home Connect Appliance(s)

Siemens CoffeeMaker TQ907DF5/01

### Log File

```shell
=================================================================================================
[4.1.2024, 07:30:08] [homebridge-homeconnect] 
[4.1.2024, 07:30:08] [homebridge-homeconnect] // Union types
[4.1.2024, 07:30:08] [homebridge-homeconnect] export type ProgramKey =
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'BSH.Common.Program.Favorite.001'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'BSH.Common.Program.Favorite.013' // (unrecognised)
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.CaffeGrande'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.CaffeLatte'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.Cappuccino'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.Coffee'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.Espresso'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.EspressoDoppio'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.EspressoMacchiato'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.HotWater'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.LatteMacchiato'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.MilkFroth'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.Ristretto'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.WarmMilk'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.Beverage.XLCoffee'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CleaningModes.ApplianceOnRinsing'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Americano'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.BlackEye'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.CafeAuLait'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.CafeConLeche'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.CafeCortado'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Cortado'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.DeadEye'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Doppio'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.FlatWhite'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Galao'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Garoto'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.GrosserBrauner'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Kaapi'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.KleinerBrauner'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.KoffieVerkeerd'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.RedEye'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.Verlaengerter'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.VerlaengerterBraun'
[4.1.2024, 07:30:08] [homebridge-homeconnect]   | 'ConsumerProducts.CoffeeMaker.Program.CoffeeWorld.WienerMelange';
[4.1.2024, 07:30:08] [homebridge-homeconnect] 
[4.1.2024, 07:30:08] [homebridge-homeconnect] // EventNotify
[4.1.2024, 07:30:08] [homebridge-homeconnect] export interface EventNotifyValues {
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'BSH.Common.Option.ProgramProgress'?:                           number;
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'BSH.Common.Root.ActiveProgram'?:                               ProgramKey; // (unrecognised)
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'BSH.Common.Root.SelectedProgram'?:                             ProgramKey; // (unrecognised)
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'BSH.Common.Setting.PowerState'?:                               PowerState;
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'ConsumerProducts.CoffeeMaker.Option.BeanContainerSelection'?:  BeanContainerSelection;
[4.1.2024, 07:30:08] [homebridge-homeconnect]     'ConsumerProducts.CoffeeMaker.Option.FillQuantity'?:            number;
[4.1.2024, 07:30:08] [homebridge-homeconnect] }
[4.1.2024, 07:30:08] [homebridge-homeconnect] 
[4.1.2024, 07:30:08] [homebridge-homeconnect] ===================================================================================
```

### Additional Information

Hallo, Yesterday after the update it works.

**Comment by thoukydides** [comment]
Thanks for the report. I should have seen that coming after #203!
I have now added programs up to `BSH.Common.Program.Favorite.020` in [v0.37.7](https://github.com/thoukydides/homebridge-homeconnect/releases/tag/v0.37.7).
Please raise a new issue (or re-open this one) if you see any more.

**Comment by Mibra-de** [comment]
It works, thank you for the very fast help
